### PR TITLE
Do not attempt to re-read an empty XPI

### LINKED ISSUE #293: Better handle empty ZIP files in `Xpi`
The problem is that for empty ZIP files, we re-open and re-read the ZIP file, except when we disable the "auto-close" feature. In this case, we do not close the ZIP file and attempt to re-read it directly, which, I _think_, is the root cause here.

~~Maybe we'd need to rewind the file descriptor somehow? Changing the line linked above would be an alternative except that it is assumed _everywhere_ that the default set of `files` is an empty dict and many test cases simply set `files` to some sets of files and have assertions on these files. We'd need to update lots of tests if we were to introduce a different prop and/or make `files` optional (`Files | null`).~~

This ^ is the right approach, let's add and use a new `processed` property on `Xpi`, and update the test suite.

_Originally posted by @willdurand in https://github.com/mozilla/addons-linter/issues/4588#issuecomment-1375362364_

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #293

### CODE REVIEW

File: src/io/xpi.ts [hunk 1]
```diff
@@ -233,4 +237,10 @@ export class Xpi extends IOBase {
       this.zipfile.close();
     }
   }
+
+  _overrideCachedFilesForTests(files: Files) {
```
Comment (diox) [inline src/io/xpi.ts:241]: I am not a big fan of exporting a method just for tests... I think I would have preferred to repeat the `myXpi.processed = true;` in all these tests instead, wdyt ?

Comment (willdurand) [reply src/io/xpi.ts:241]: This is future-proofing in case we need to make more changes in the future.. The alternative would be a sub-class for all tests only defined in the test files, maybe? I would like to avoid repeating `.processed = true` everywhere (in addition to `.files = { ... }`).

Comment (diox) [reply src/io/xpi.ts:241]: I think I prefer that

Comment (willdurand) [reply src/io/xpi.ts]: done

**Review by diox — APPROVED** [review]
